codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessageDecoder.java

```diff
@@ -357,4 +357,36 @@ private void fail(final ChannelHandlerContext ctx, String errMsg, Throwable t) {
         }
         throw ppex;
     }
+
+    /**
+     * Returns {@code true} if the given {@link ByteBuf} is a valid HAProxy header.
+     * Be aware that this method will not increase the readerIndex of the given {@link ByteBuf}.
+     *
+     * @param buffer
+     *                  The {@link ByteBuf} to read from. Be aware that it must have at least 12 bytes to read,
+     *                  otherwise it will throw an {@link IllegalArgumentException}.
+     * @return isValid
+     *                  {@code true} if the {@link ByteBuf} is a valid HAProxy header, {@code false} otherwise.
+     * @throws IllegalArgumentException
+     *                  Is thrown if the given {@link ByteBuf} has not at least 12 bytes to read.
+     */
+    public static boolean isValidHeader(final ByteBuf buffer) {
```

Comment:
I just did the same as we de in SslHandler here. So it's consistent. WDYT?



> Am 04.06.2015 um 06:59 schrieb Trustin Lee <notifications@github.com>:
> 
> In codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessageDecoder.java:
> 
> > @@ -357,4 +357,36 @@ private void fail(final ChannelHandlerContext ctx, String errMsg, Throwable t) {
> >          }
> >          throw ppex;
> >      }
> > +
> > +    /**
> > +     * Returns {@code true} if the given {@link ByteBuf} is a valid HAProxy header.
> > +     * Be aware that this method will not increase the readerIndex of the given {@link ByteBuf}.
> > +     *
> > +     * @param buffer
> > +     *                  The {@link ByteBuf} to read from. Be aware that it must have at least 12 bytes to read,
> > +     *                  otherwise it will throw an {@link IllegalArgumentException}.
> > +     * @return isValid
> > +     *                  {@code true} if the {@link ByteBuf} is a valid HAProxy header, {@code false} otherwise.
> > +     * @throws IllegalArgumentException
> > +     *                  Is thrown if the given {@link ByteBuf} has not at least 12 bytes to read.
> > +     */
> > +    public static boolean isValidHeader(final ByteBuf buffer) {
> How about returning Boolean and return null when there are not enough bytes? Otherwise, the caller must have the prior knowledge about the required number of readable bytes.
> 
> —
> Reply to this email directly or view it on GitHub.
>